internal/kms/pubsub.go

```diff
@@ -36,13 +40,20 @@ type PubSubServer interface {
 	SendPubSubMessage(context.Context, string, []byte) (<-chan rpc.Response, error)
 }
 
+type CollectionRetriever interface {
```

Comment:
question: Does this need to be interfaced?  There is only one type that implements it.

nitpick: The concrete type is a pointer to a pointer, I understand that is driven by Go norms, but it still feels a little unnecessary when I see it :)

Reply:
This interface allows KMS to depend on only `client` type instead of `db`

the concrete type is `*db.CollectionRetriever`. I don't see 2 pointes there.

Reply:
The only thing `*db.CollectionRetriever` contains is a pointer to `db` - thus all instances of `kms.CollectionRetriever` are pointers to a pointer.